ee/src/main/java/org/jboss/as/ee/security/JaccEarDeploymentProcessor.java

```diff
@@ -27,10 +30,9 @@
  */
 public class JaccEarDeploymentProcessor implements DeploymentUnitProcessor {
 
-    private final String jaccCapabilityName;
+    private volatile boolean deployed = false;
```

Comment:
This needs a usage comment, because it's fragile. Maybe it needs a rename?

The instance of this class is not bound to a deployment, so any state it holds is globally applicable. But 'deployed' intuitively means something applicable to a particular deployment.

Maybe something like

// Tracks whether we've detected that one of the capabilities indicating a need for Jakarta Authorization is present
private volatile boolean detectedJakartaAuthentication = false;

The presence of absence of the capability won't change without a reload (otherwise this field isn't valid at all), so you could also change the 'deploy' impl to just skip looking for the capabilities if this is 'true'.

Reply:
+1. 

AbstractSecurityDeployer and subclasses should be renamed. They are AbstractJaccServiceFactory etc and have a 'createJaccService" method, not a 'deploy' method. They don't deploy anything in any sense other than adding an object to the heap.

That doesn't need to be in scope for this PR though.

Reply:
+1 I would actually like to see what we can pull out here, there is a little but too much code IMO in the WildFly repo.

If we could refactor and pull some out it would makeit easier to use with Undertow standalone as well which wouild also make developing and testing this a lot easier.